Knoema.Client/Client.cs

```diff
@@ -629,5 +629,49 @@ public Task<IReadOnlyList<CompanyModel>> GetCompanies()
 		{
 			return ApiGet<IReadOnlyList<CompanyModel>>("/api/1.0/meta/tickers");
 		}
+
+		public Task<int> GetSeriesCount(string datasetId)
+		{
+			return ApiPost<int>($"/api/1.1/data/dataset/{datasetId}/count", new StringContent(JsonConvert.SerializeObject(
+				new 
+				{ 
+					IncludeUnitsInfo = false 
+				}), 
+				Encoding.UTF8, 
+				"application/json"));
+		}
+
+		public async Task CreateReplacement(string originalDatasetId, string replacementDatasetId)
+		{
+			var message = new HttpRequestMessage(HttpMethod.Post, GetUri(_host, $"/api/1.0/meta/dataset/{originalDatasetId}"))
+			{
+				Content = new StringContent(JsonConvert.SerializeObject(
+					new
+					{
+						ReplacementDatasetId = replacementDatasetId
+					}), Encoding.UTF8, "application/json")
+			};
+			var response = await ProcessRequest(message);
+			response.EnsureSuccessStatusCode();
+			
+			var responseContent = await response.Content.ReadAsStringAsync();
+			if (!string.IsNullOrEmpty(responseContent))
+			{
+				var resultStatus = new ResultStatusViewModel();
+
+				try
+				{
+					resultStatus = JsonConvert.DeserializeObject<ResultStatusViewModel>(responseContent);
+				}
+				catch { }
```

Comment:
I think you should handle this exception and raise a new one.

Reply:
I also have a question here what if a new field appears in settings? How will it work. And i agree with Misha it is a bad approach to hide exceptions